tests/trainer/test_trainer.py

```diff
@@ -1,247 +1,375 @@
+        return {
+            'model': SimpleConvModel(),
+            'train_dataloader': DataLoader(
+                dataset=RandomImageDataset(size=16),
+                batch_size=4,
+            ),
+            'eval_dataloader': None,
+            'max_duration': '1ep',
+            'loggers': []
+        }
+
+    def test_data_augmented(self, config):
+        config['algorithms'] = [CutOut(n_holes=1, length=5)]
+
+        # we give the callback access to the dataset to test
+        # that the images have been augmented.
+        config['callbacks'] = [
+            AssertDataAugmented(dataset=config['train_dataloader'].dataset),
+        ]
+        trainer = Trainer(**config)
+        trainer.fit()
+
+    def test_data_not_augmented(self, config):
+        config['callbacks'] = [
+            AssertDataAugmented(dataset=config['train_dataloader'].dataset),
+        ]
+        trainer = Trainer(**config)
+        with pytest.raises(AssertionError):
+            trainer.fit()
+
+
+"""
+The below is a catch-all test that runs the Trainer
+with each algorithm, callback, and loggers. Success
+is defined as a successful training run.
+
+This should eventually be replaced by functional
+tests for each object, in situ of our trainer.
+
+We use the hparams_registry associated with our
+config management to retrieve the objects to test.
+"""
+
+
+class TestTrainerAssets:
+
+    @pytest.fixture
+    def config(self):
+        return {
+            'model': SimpleConvModel(),
+            'train_dataloader': DataLoader(
+                dataset=RandomImageDataset(size=16),
+                batch_size=4,
+            ),
+            'eval_dataloader': DataLoader(
+                dataset=RandomImageDataset(size=16),
+                batch_size=4,
+            ),
+            'max_duration': '2ep',
+            'loggers': [],  # no progress bar
+        }
+
+    # Note: Not all algorithms, callbacks, and loggers are compatible
+    #       with the above configuration. The fixtures below filter and
+    #       create the objects to test.
+
+    @pytest.fixture(params=algorithms_registry.items(), ids=algorithms_registry.keys())
+    def algorithm(self, request):
+
+        name, hparams = request.param
+        skip_list = {
+            'swa': 'SWA not compatible with composed schedulers.',
+            'alibi': 'Not compatible with conv model.',
+            'seq_length_warmup': 'Not compatible with conv model.',
+            'randaugment': 'Requires PIL dataset to test.',
+            'augmix': 'Required PIL dataset to test.',
+            'stochastic_depth': 'Only applies to ResNets.',
+            'no_op_model': 'Not compatible with this model.'
+        }
+
+        # skip any tests incompatible with this config
+        if name in skip_list:
+            pytest.skip(skip_list[name])
+        elif name in ('cutmix, mixup, label_smoothing'):
+            pytest.importorskip("torch", minversion="1.10", reason="Pytorch 1.10 required.")
```

Comment:
Can you add more context / perhaps link to https://github.com/mosaicml/composer/issues/362 in a comment to why they require 1.10?